test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/YearTest.scala

```diff
@@ -0,0 +1,415 @@
+      expectThrows(classOf[DateTimeException], t.atDay(-1))
+    }
+
+    assertEquals(LocalDate.of(leapYear.getValue, 2, 29), leapYear.atDay(31 + 29))
+    assertEquals(LocalDate.of(leapYear.getValue, 3, 1), leapYear.atDay(31 + 30))
+    assertEquals(LocalDate.of(leapYear.getValue, 12, 30), leapYear.atDay(365))
+    assertEquals(LocalDate.of(leapYear.getValue, 12, 31), leapYear.atDay(366))
+    expectThrows(classOf[DateTimeException], leapYear.atDay(367))
+
+    assertEquals(LocalDate.of(nonLeapYear.getValue, 3, 1), nonLeapYear.atDay(31 + 29))
+    assertEquals(LocalDate.of(nonLeapYear.getValue, 12, 31), nonLeapYear.atDay(365))
+    expectThrows(classOf[DateTimeException], nonLeapYear.atDay(366))
+  }
+
+  @Test def isLeap(): Unit = {
+    assertTrue(leapYear.isLeap)
+    assertFalse(nonLeapYear.isLeap)
+
+    assertTrue(Year.isLeap(leapYear.getValue))
+    assertFalse(Year.isLeap(nonLeapYear.getValue))
+  }
+
+  @Test def isValidMonthDay(): Unit = {
+    val leapMonthDay = MonthDay.of(2, 29)
+    assertTrue(leapYear.isValidMonthDay(leapMonthDay))
+    assertFalse(nonLeapYear.isValidMonthDay(leapMonthDay))
+
+    for {
+      t <- samples
+      monthDay <- monthDaySamples
+    } {
+      assertTrue(t.isValidMonthDay(monthDay))
+    }
+  }
+
+  @Test def atMonthDay(): Unit = {
+    val leapMonthDay = MonthDay.of(2, 29)
+    val leapDate = LocalDate.of(leapYear.getValue(),
+        leapMonthDay.getMonthValue, leapMonthDay.getDayOfMonth)
+    assertEquals(leapDate, leapYear.atMonthDay(leapMonthDay))
+
+    val nonLeapDate = LocalDate.of(nonLeapYear.getValue(),
+        leapMonthDay.getMonthValue, 28)
+    assertEquals(nonLeapDate, nonLeapYear.atMonthDay(leapMonthDay))
+
+    for {
+      t <- samples
+      monthDay <- monthDaySamples
+    } {
+      val expected = LocalDate.of(t.getValue, monthDay.getMonthValue, monthDay.getDayOfMonth)
+      assertEquals(expected, t.atMonthDay(monthDay))
+    }
+  }
+
+  @Test def compareTo(): Unit = {
+    assertEquals(0, min.compareTo(min))
+    assertEquals(-1, Math.signum(min.compareTo(max)).toInt)
+    assertEquals(1, Math.signum(max.compareTo(min)).toInt)
+    assertEquals(0, max.compareTo(max))
+  }
+
+  @Test def isAfter(): Unit = {
+    assertFalse(min.isAfter(min))
+    assertFalse(min.isAfter(max))
+    assertTrue(max.isAfter(min))
+    assertFalse(max.isAfter(max))
+  }
+
+  @Test def isBefore(): Unit = {
+    assertFalse(min.isBefore(min))
+    assertTrue(min.isBefore(max))
+    assertFalse(max.isBefore(min))
+    assertFalse(max.isBefore(max))
+  }
+
+  @Test def now(): Unit = {
+    val now = LocalDate.now()
+    val year = Year.now()
+
+    if (now.getYear != year.getValue) {
+      assertEquals(LocalDate.now().getYear, year.getValue)
+      println("Happy new year " + year + '!')
+    }
+  }
+
+  @Test def of(): Unit = {
+    assertEquals(lastBCYear, Year.of(0))
+    assertEquals(firstACYear, Year.of(1))
+    assertEquals(min, Year.of(MIN_VALUE))
+    assertEquals(max, Year.of(MAX_VALUE))
```

Comment:
Should also test the contents of the year that is returned by `of`. Because if for some reason `Year.of(0)` would return `Year(5)` these tests would still pass.